### PR TITLE
chore: Added e2e for  hooks

### PR DESCRIPTION
**Pull request by sarabala1979**
Signed-off-by: Saravanan Balasubramanian <sarabala1979@gmail.com>

Fixes #TODO

### CODE REVIEW

File: test/e2e/hooks_test.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      template: http
+  templates:
+    - name: main
+      steps:
+      - - name: step1
+          template: http
+
+    - name: http
+      http:
+        # url: http://dummy.restapiexample.com/api/v1/employees
```
Comment (terrytangyuan) [inline test/e2e/hooks_test.go:43]: Remove this?

File: test/e2e/hooks_test.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      steps:
+      - - name: step1
+          template: http
+
+    - name: http
+      http:
+        url: "https://raw.githubusercontent.com/argoproj/argo-workflows/4e450e250168e6b4d51a126b784e90b11a0162bc/pkg/apis/workflow/v1alpha1/generated.swagger.json"
+`).When().
+		SubmitWorkflow().
+		WaitForWorkflow().
```
Comment (alexec) [inline test/e2e/hooks_test.go:48]:
```
WaitForWorkflow(fixtures.ToBeSucceeded).
```

File: test/e2e/hooks_test.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+              success:
+                expression: steps["step-1"].status == "Succeeded"
+                template: http
+            template: http
+    - name: http
+      http:
+        url: "https://raw.githubusercontent.com/argoproj/argo-workflows/4e450e250168e6b4d51a126b784e90b11a0162bc/pkg/apis/workflow/v1alpha1/generated.swagger.json"
+`).When().
+		SubmitWorkflow().
+		WaitForWorkflow().
```
Comment (alexec) [inline test/e2e/hooks_test.go:88]: as above

File: test/e2e/hooks_test.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+		Then().
+		ExpectWorkflow(func(t *testing.T, metadata *v1.ObjectMeta, status *v1alpha1.WorkflowStatus) {
+			assert.Equal(t, status.Phase, v1alpha1.WorkflowSucceeded)
+		}).ExpectWorkflowNode(func(status v1alpha1.NodeStatus) bool {
+		if strings.Contains(status.Name, "hook") {
+			return true
+		}
+		return false
+	}, func(t *testing.T, status *v1alpha1.NodeStatus, pod *v12.Pod) {
```
Comment (alexec) [inline test/e2e/hooks_test.go:57]: I think you need to fix this import alias

File: test/e2e/hooks_test.go [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+kind: Workflow
+metadata:
+  generateName: lifecycle-hook-
+spec:
+  entrypoint: main
+  hooks:
+    exit:
+      template: http
+    running:
+      expression: workflow.status == "Running"
```
Comment (alexec) [inline test/e2e/hooks_test.go:35]: interesting - this fires every time? I'm actualy not sure how to interpret this hook.

Comment (sarabala1979) [reply test/e2e/hooks_test.go:35]: it will fire only one time the first time of expression satisfy.

**Pull request merged**